src/vs/workbench/contrib/notebook/browser/notebookEditor.ts

```diff
@@ -231,24 +234,31 @@ export class NotebookEditor extends EditorPane implements INotebookEditorPane {
 			if (model === null) {
 				const knownProvider = this._notebookService.getViewTypeProvider(input.viewType);
 
-				if (knownProvider) {
-					throw createEditorOpenError(new Error(localize('fail.noEditor', "Cannot open resource with notebook editor type '{0}', please check if you have the right extension installed and enabled.", input.viewType)), [
-						toAction({
-							id: 'workbench.notebook.action.installMissing', label: localize('notebookOpenInstallMissingViewType', "Install extension for '{0}'", input.viewType), run: async () => {
-								const d = this._notebookService.onAddViewType(viewType => {
-									if (viewType === input.viewType) {
-										// serializer is registered, try to open again
-										this._editorService.openEditor({ resource: input.resource });
-										d.dispose();
-									}
-								});
-								await this._instantiationService.createInstance(InstallRecommendedExtensionAction, knownProvider).run();
-							}
-						})
-					], { allowDialog: true });
-				} else {
+				if (!knownProvider) {
 					throw new Error(localize('fail.noEditor', "Cannot open resource with notebook editor type '{0}', please check if you have the right extension installed and enabled.", input.viewType));
 				}
+
+				throw createEditorOpenError(new Error(localize('fail.noEditor.extensionMissing', "Cannot open resource with notebook editor type '{0}', please check if you have the right extension installed and enabled.", input.viewType)), [
+					toAction({
+						id: 'workbench.notebook.action.installMissing', label: localize('notebookOpenInstallMissingViewType', "Install extension for '{0}'", input.viewType), run: async () => {
```

Comment:
The text here says "Install extension for", but below, it looks like we only try to enable the extension. Does the setEnablement line also install the extension, or is the extension installed elsewhere?

Reply:
Updated the wording to "Install and enable". This can be known when creating the actions as extensions might not be loaded yet on window reload so it's only known when users click on the button later.